STYLE_GUIDE.adoc

```diff
@@ -3,28 +3,38 @@
 
 toc::[]
 
-= File naming conventions
+= Default conventions
 
-[source,typescript]
-----
-// bad
-import Footer from './Footer/index';
-// good
-import Footer from './Footer';
-----
+By default, we try to follow community conventions. link:https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md[Basarat's Style Guide] is used as a reference, with a few exceptions that are listed below:
+
+== File names
 
+Most files are named after the main type / class / interface it's exporting. For instance, `ServiceList.ts` is named after the `ServiceList` interface that it's exporting. Which means that, unlike in Basarat's guide, we often use `PascalCase`.
 
-= Methods
+Files with more general purpose, or the ones that don't export types or such, are `camelCase` (ex: `routes.ts`).
 
-=== Naming
+== Variables, functions, constants
 
-* action types - use constants, [NOUN]_[VERB], eg. USER_FETCH or MATERIAL_REQUEST_UPDATE
-* action creators - [verb][Noun]() eg. fetchCurrentUser(), acceptMaterialRequest()
-* selectors - get[NOUN]() or select[NOUN]() eg. getSelectedCards(), selectActiveRows()
+We mostly follow Basarat's guide, so `camelCase`. Exceptions are:
 
-=== Handlers
+- Redux actions, which are technically constants, are `PascalCase` (ex: `GraphActions`).
```

Comment:
For a follow up: Even if Redux actions are declared with a `const` keyword and, then, in an object; they follow the JS way of namespacing. So, for Redux actions, the namespace convention should apply (which we already comply)

Reply:
I don't know really about the "JS way of namespacing", basically any variable can contain functions and hence be considered as something like a namespace, no?
Typescriptly speaking it's not a namespace so I think it's good to mention.

Reply:
Yes, if we weren't using typescript, we would be creating variables that contains functions to do namespacing.
Check this simplistic TS example: https://bit.ly/2X2q0xq. You can see that, indeed, a "typescript namespace" is transpiled to a JS code that results in creating a simple variable with functions. 

I haven't tried, but probably we could convert Redux actions to code using typescript namespaces and, probably, everything may still would work without changes. If this is true, we could convert the Redux actions code to avoid this kind of exceptions to the coding style.